### ISSUE TITLE
Same Keras net gives different inference times 

### ISSUE THREAD

**Issue by MaximumProgrammer** [keras-team/keras#11763]
Hello,
i trained a keras net for different problems but with the same structure, the main thing is that even with the same net input size and same net structure after the training both nets have different weight sizes and different inference times, but the difference between both nets is just a facor of 10 for the inference time and a factor of 2 for the weight size. Where does it come from and are there ways to improve the inference time without losing any accurarcy. 

Best regards.

**Comment by ZeroCool11000** [comment]
Ok the backend is just CNTK.

**Comment by gogasca** [comment]
Can you share some code?

**Comment by ZeroCool11000** [comment]
Ok i will going for it

**Comment by ZeroCool11000** [comment]
The difference for inference time mainly comes from batch normalization, without normaliziation the net ist just faster does Keras remove batch normalization after training ?

**Comment by ZeroCool11000** [comment]
Maybe is there a way to remove batch normalization after training ?

**Comment by LinusSte** [comment]
Any news on this? Im currently having a related issue, I need to understand why Batch Normalization Layers significantly increases the inference times. Please have a look at my SO question: https://stackoverflow.com/questions/56565998/how-to-calculate-the-number-of-multiplications-happening-in-batchnormalization-l

**Comment by ZeroCool11000** [comment]
I also think you have to be care of how the net is defined.
